ui/appui-react/src/appui-react/frontstage/FrontstageConfig.ts

```diff
@@ -0,0 +1,73 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Frontstage
+ */
+
+import { CommonProps } from "@itwin/core-react";
+import { ContentGroup, ContentGroupProvider } from "../content/ContentGroup";
+import { ToolItemDef } from "../shared/ToolItemDef";
+import { PanelConfig } from "../stagepanels/PanelConfig";
+import { WidgetConfig } from "../widgets/WidgetConfig";
+
+/** Configuration from which a frontstage is created.
+ * @public
+ */
+export interface FrontstageConfig extends CommonProps {
+  /** Id for the Frontstage */
+  readonly id: string;
+  /** Tool that is started once the Frontstage is activated */
+  readonly defaultTool: ToolItemDef;
+  /** The Content Group providing the Content Views */
+  readonly contentGroup: ContentGroup | ContentGroupProvider;
+  /** Id of the Content View to be activated initially */
+  readonly defaultContentId?: string;
+  /** Any application data to attach to this Frontstage. */
+  readonly applicationData?: any;
+  /** Usage type for this Frontstage. */
+  readonly usage?: string;
+  /** Frontstage version. Used to force saved layout reinitialization after changes to frontstage.
+   * @note This value should be increased when changes are made to Frontstage.
+   * Increasing the value will make sure to reinitialize App layout instead of restoring to old layout.
+   * Version increase is required when widgets are added/removed.
+   */
+  readonly version?: number;
+  /** if isIModelIndependent then frontstage is independent from any iModel. */
+  readonly isIModelIndependent?: boolean;
```

Comment:
i wasnt aware we had this; what is the use case for this?

Reply:
`version` is to invalidate the saved/chached frontstage version.
Not sure what the use-case is for `isIModelIndependent` (added by https://github.com/iTwin/itwinjs-core/pull/2423), @NancyMcCallB?

Reply:
imo these should never have been inside a stage; so good thing we're removing it